conbench/entities/history.py

```diff
@@ -570,3 +582,67 @@ def _calculate_z_score(
         z_score = z_score * -1
 
     return z_score
+
+
+def zscore_automated_detect(df: pd.DataFrame, z_score_threshold=5.0) -> pd.DataFrame:
+    """Detect outliers and distribution shifts in historical data
```

Comment:
:heart: I like this sentence. I didn't know that we wanted to automatically detect distribution shifts, too. Is this required for the outlier detection? Sorry if that should be clear by now, but I'd appreciate clarification.